root/user/UserProfile.js

```diff
@@ -0,0 +1,515 @@
+  const typesList = [];
+  if (user.deleted) {
+    typesList.push(l('Deleted User'));
+  }
+  if (user.is_auto_editor) {
+    typesList.push(exp.l('{doc|Auto-Editor}', {doc: '/doc/Editor'}));
+  }
+  if (user.is_bot) {
+    typesList.push(l('Internal/Bot'));
+  }
+  if (user.is_relationship_editor) {
+    typesList.push(exp.l('{doc|Relationship Editor}', {doc: '/doc/Editor'}));
+  }
+  if (user.is_wiki_transcluder) {
+    typesList.push(exp.l('{doc|Transclusion Editor}', {doc: '/doc/Editor'}));
+  }
+  if (user.is_location_editor) {
+    typesList.push(exp.l('{doc|Location Editor}', {doc: '/doc/Editor'}));
+  }
+  if (user.is_limited) {
+    typesList.push(
+      <span
+        className="tooltip"
+        title={l(
+          `User accounts must be more than 2 weeks old, have a confirmed email address,
+           and more than 10 accepted edits in order to vote on others' edits.`,
+        )}
+      >
+        {l('Beginner')}
+      </span>,
+    );
+  }
+  // If no other types apply, then this is a normal user
+  if (typesList.length === 0) {
+    typesList.push(l('Normal User'));
+  }
+
+  return typesList;
+}
+
+type UserProfilePropertyProps = {|
+  +children: React.Node,
+  +className?: string,
+  +name: string,
+|};
+
+const UserProfileProperty = ({
+  children,
+  className,
+  name,
+}: UserProfilePropertyProps) => (
+  <tr className={className || null}>
+    <th>{name}</th>
+    <td>{children}</td>
+  </tr>
+);
+
+type UserProfileInformationProps = {|
+  +$c: CatalystContextT,
+  +subscribed: boolean,
+  +subscriberCount: number,
+  +user: EditorT,
+  +viewingOwnProfile: boolean,
+|};
+
+const UserProfileInformation = withCatalystContext(({
+  $c,
+  subscribed,
+  subscriberCount,
+  user,
+  viewingOwnProfile,
+}: UserProfileInformationProps) => {
+  const showBioAndURL = !user.is_limited || $c.user_exists;
+  let memberSince;
+  if (user.name === 'rob') {
+    memberSince = l('The Dawn of the Project');
+  } else if (user.is_charter) {
+    memberSince = l('The early days of the project');
+  } else {
+    memberSince = formatUserDate($c.user, user.registration_date);
+  }
+
+  const noEmailWarning = (viewingOwnProfile &&
+    $c.user && !$c.user.has_confirmed_email_address) ? (
+      <strong>
+        {l(`Your homepage and biography will not show
+            until you have confirmed your email.`)}
+      </strong>
+    ) : null;
+
+  return (
+    <>
+      <h2>{l('General Information')}</h2>
+
+      <table className="profileinfo" role="presentation">
+        <UserProfileProperty name={l('Email:')}>
+          {user.email ? (
+            <>
+              {viewingOwnProfile ? user.email : l('(hidden)')}
+              {' '}
+              {user.email_confirmation_date ? (
+                exp.l('(verified at {date})',
+                      {
+                        date: formatUserDate(
+                          $c.user, user.email_confirmation_date,
+                        ),
+                      })
```

Comment:
The whitespace looks a little excessive compared to something like
```JavaScript
                exp.l('(verified at {date})', {
                  date: formatUserDate($c.user, user.email_confirmation_date),
                })
```

Reply:
I thought we wanted date to be at the same indentation as the string. But ok, made the change.

Reply:
Yeah, that's true, If every argument is on a separate line they should line up. Just not sure we need to force them onto separate lines in all cases.